db_setup/app/__init__.py

```diff
@@ -16,63 +17,137 @@
 
 
 def import_resources():
+    # Step 1: Get data
     with open('resources.yml', 'r') as f:
         data = yaml.load(f)
 
-    resource_dict = {}
-    language_dict = {}
-    category_dict = {}
-    unique_resources = []
+    # Step 2: Uniquify resources
+    unique_resources = remove_duplicates(data) 
 
-    for resource in data:
-        if not resource_dict.get(resource['url']):
-            resource_dict[resource['url']] = True
-            unique_resources.append(resource)
+    # Step 3: Get existing entries from DB
+    try:
+        resources_list = Resource.query.all()
+        languages_list = Language.query.all()
+        categories_list = Category.query.all()
 
+        # Convert to dict for quick lookup
+        existing_resources = {r.url:r for r in resources_list}
```

Comment:
This isn't bad if you need to preserve the actual item, which you are. You could give give each model class a method 
```
def key(self):
    return self.name  # or self.url
```
then here you'd call 
` {key(l) : l for l in languages_list } `